sdk/core/core/src/az_version.c

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_version.h"
+
+int az_core_version_get_major() { return AZ_VERSION_MAJOR; }
```

Comment:
You have functions that return the symbols. Can we kill the functions, and just use the symbols?

Reply:
The version comes from a file.  We could move it to be a #define which the build checks out and replaces.
Something like:
`#define AZ_VERSION_STRING 1.0.0-preview.1`

As for using the defines directly, We could do that, If we do that then there are two options for using the compiler passed in options.
Option 1:
Compiler define which is redefined to our internal in source.  Drawback is multiple defines of the same thing.
```
// cl -DBUILD_VERSION_MAJOR=1
#define AZ_VERSION_MAJOR BUILD_VERSION_MAJOR

```
Option 2:
Define at compile time and do nothing in code.  In that case version.h is just an empty file with comments to help folks discover what they should be doing.  This approach probably suffers with code completions as there is no way to attach the doxygen comments to the "thing".
```
//The below defines come from the build, 
//  AZ_VERSION_MAJOR
//  AZ_VERSION_MINOR
//  AZ_VERSION_PATCH
```

Reply:
I think we lose intellisense by using only the symbols right?  Unless we make a second #define on az_version.h
I don't know a way to set up intellisense to consider symbols from build.

I would leave this functions better.
I believe functions give us more flexibility for the future than just the symbols.  We could have IN_LINE functions in header file maybe.